doc/guides/dts/index.rst

```diff
@@ -346,44 +346,389 @@ Below is sample header content generated for this node, in
+| Type             | Example                | Generated macros                       |
++==================+========================+========================================+
+| ``int``          | ``foo = <1>``          | ``#define DT_<node>_FOO 1``            |
++------------------+------------------------+----------------------------------------+
+| ``array``        | ``foo = <1 2>``        | | ``#define DT_<node>_FOO_0 1``        |
+|                  |                        | | ``#define DT_<node>_FOO_1 2``        |
+|                  |                        | | ``#define DT_<node>_FOO {1, 2}``     |
++------------------+------------------------+----------------------------------------+
+| ``string``       | ``foo = "bar"``        | ``#define DT_<node>_FOO "bar"``        |
++------------------+------------------------+----------------------------------------+
+| ``string-array`` | ``foo = "bar", "baz"`` | | ``#define DT_<node>_FOO_0 "bar"``    |
+|                  |                        | | ``#define DT_<node>_FOO_1 "baz"``    |
++------------------+------------------------+----------------------------------------+
+| ``uint8-array``  | ``foo = [01 02]``      | ``#define DT_<node>_FOO {0x01, 0x02}`` |
++------------------+------------------------+----------------------------------------+
+
+For ``type: boolean``, the generated macro is set to 1 if the property exists
+on the node, and to 0 otherwise:
 
 .. code-block:: none
 
-   aliases {
-   	i2c-1 = &i2c;
+   #define DT_<node>_FOO 0/1
+
+The generation for ``type: phandle-array`` is the most complex. To understand
+it, it is a good idea to first go through the documentation for
+``phandle-array`` in :ref:`bindings`.
+
+Take the following devicetree nodes and binding contents as an example:
+
+.. code-block:: none
+   :caption: Devicetree nodes for PWM controllers
+
+   pwm-ctrl-0: pwm-controller-0 {
+   	compatible = "vendor,pwm-controller";
+   	label = "PWM_0";
+   	#pwm-cells = <2>;
+   	...
+   };
+
+   pwm-ctrl-1: pwm-controller-1 {
+   	compatible = "vendor,pwm-controller";
+   	label = "PWM_1";
+   	#pwm-cells = <2>;
+   	...
    };
 
-This would generate additional ``DT_ALIAS_I2C_1_...`` aliases for all
-properties in the output for the node with the devicetree label ``i2c``.
+.. code-block:: yaml
+   :caption: ``pwm-cells`` declaration in binding for ``vendor-pwm-controller``
+
+   pwm-cells:
+       - channel
+       - period
+
+Assume the property assignment looks like this:
+
+.. code-block:: none
+
+   pwms = <&pwm-ctrl-0 1 10
+           &pwm-ctrl-1 2 20>;
+
+These macros then get generated:
+
+.. code-block:: none
+
+   #define DT_<node>_PWMS_CONTROLLER_0    "PWM_0"
+   #define DT_<node>_PWMS_CHANNEL_0       1
+   #define DT_<node>_PWMS_PERIOD_0        10
+
+   #define DT_<node>_PWMS_CONTROLLER_1    "PWM_1"
+   #define DT_<node>_PWMS_CHANNEL_1       2
+   #define DT_<node>_PWMS_PERIOD_1        20
+
+   /* Initializers */
```

Comment:
Do we have support for that in the tree? I thought it will be implemented in #21393

Reply:
Yup, already supported.

Honestly, I'm just reverse-engineering this stuff from the code as I go. Even though I wrote most of it, it's hard for me to keep track of how all of it works, because it just emulates the logic in the old scripts.